### PR TITLE
refactor(x/mint): initialize inflation decimals once

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/2501

### CODE REVIEW

**Review by SweeXordious — DISMISSED** [review]

File: x/mint/types/constants.go [hunk 1]
```diff
@@ -19,14 +19,8 @@ const (
...
-}
-
-func TargetInflationRateAsDec() sdk.Dec {
-	return sdk.NewDecWithPrec(TargetInflationRate*1000, 3)
-}
+var (
+	InitialInflationRateAsDec = sdk.NewDecWithPrec(InitialInflationRate*1000, 3)
+	DisinflationRateAsDec     = sdk.NewDecWithPrec(DisinflationRate*1000, 3)
+	TargetInflationRateAsDec  = sdk.NewDecWithPrec(TargetInflationRate*1000, 3)
+)
```
Comment (evan-forbes) [inline x/mint/types/constants.go:26]: I don't care that much, but could we allocate once as an unexported global var, then use a function to enforce that they are only ever read from? global vars are the devil

**Review by evan-forbes — APPROVED** [review]

**Comment by staheri14** [comment]
> Addresses finding 005 of binary builders preliminary audit report.

Is there any GH issue for their finding?

**Review by staheri14 — DISMISSED** [review]
LGTM! I'm not entirely certain about the reasoning behind this change (as no GH issue is linked), but the previous approach offers improved encapsulation, especially if these variables might need to be generated from a more intricate logic in the future.

**Comment by rootulp** [comment]
> Is there any GH issue for their finding?

No they didn't create GH issues for their findings. IDK why not. They shared their preliminary report in Slack so I just DM'ed it to you.

**Comment by staheri14** [comment]
> > Is there any GH issue for their finding?
> 
> No they didn't create GH issues for their findings. IDK why not. They shared their preliminary report in Slack so I just DM'ed it to you.

Thanks for DM! I think if we could provide some context about the motivation of the PR changes in the PR description, it makes PRs more self-contained and easier to review, wdyt?

File: x/mint/types/constants.go (hunk 1 above)
Comment (rootulp) [reply x/mint/types/constants.go]: Resolved via dde51c8da0eeab38d96d58d359618c391a55a222

**Comment by rootulp** [comment]
Agreed the normal workflow should be PR closes GH issue and the GH issue should contain context to make it easy to review.

 IDK why Binary Builders didn't create GH issues for their findings but I intentionally didn't create GH issues b/c AFAIK their report isn't public.

**Review by staheri14 — APPROVED** [review]

**Comment by rootulp** [comment]
Binary Builders created GH issues so I updated all the PRs that close Binary Builders audit findings w/ links to the GH issues (including this one).

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**